expand/expand.go

```diff
@@ -835,6 +842,28 @@ func (cfg *Config) glob(base, pat string) ([]string, error) {
 	return matches, nil
 }
 
+// stat calls cfg.Stat if it's set, otherwise calls cfg.ReadDir on name's
+// parent directory, and returns the entry for name.
+func (cfg *Config) stat(name string) (os.FileInfo, error) {
+	if cfg.Stat != nil {
+		return cfg.Stat(name)
+	}
+
+	dir, basename := filepath.Split(name)
```

Comment:
Trying to think if there's any edge case we need to worry about for `filepath.Split`, like `/`. I think not, because the shortest globs possible are `*`, which will use PWD, and `/*`, where we'll start the globbing after the root as per `IsAbs` above.

I would write a brief bit of docs here, along the lines of "we don't need to worry about name being the root of the filesystem".